hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/metrics/FederationRPCPerformanceMonitor.java

```diff
@@ -153,7 +157,7 @@ public void proxyOpComplete(boolean success, String nsId,
     if (success) {
       long proxyTime = getProxyTime();
       if (proxyTime >= 0) {
-        if (metrics != null) {
+        if (metrics != null && !nsId.equals(CONCURRENT)) {
           metrics.addProxyTime(proxyTime, state);
         }
         if (nameserviceRPCMetricsMap != null &&
```

Comment:
In the unit tests, can you also check that the proxy time for the "concurrent" namespace is being created.